### DISCUSSION TITLE
Is it possible to detect math environments in markdown files?

### DISCUSSION THREAD

**Discussion by medwatt** [nvim-treesitter/nvim-treesitter#2907] [Ideas]
I use markdown to write my notes these days and I also tend to write equations. These are defined exactly as they are in LaTex:

- inline equations are enclosed within `$ $`
- display equations are enclosed within `\[ \]`

It would be nice if these could be added to the markdown treesitter.

**Comment by theHamsta** [comment]
Such an addition would require an extension of https://github.com/MDeiml/tree-sitter-markdown because otherwise we would have to parse the enviroments in our queries which is more difficult. Maybe you can ask at that repo whether that feature is in scope for that parser project.

**Reply by theHamsta** [reply]
Yes, but in principle it would be also possible to run regexes over the markdown to find the `$` pairs but you will have the same difficulties to find the correct matching pairs like in traditional regex highlighting
